### PR TITLE
[Fix #495] Default rails console strategy configuration

### LINKED ISSUE #495: Avoiding :urgent strategy in the console?
In some of our staging environments we don't have ES hooked up, and it's controlled by an environment variable to set all strategies to `:bypass` in an initializer. However when you fire up a console in that environment, an instance of the `:urgent` strategy gets added to the stack:

```
irb(main):003:0> Chewy.root_strategy
=> :bypass
irb(main):004:0> Chewy.request_strategy
=> :bypass
irb(main):005:0> Chewy.strategy
=> #<Chewy::Strategy:0x00000008545698 @stack=[#<Chewy::Strategy::Base:0x00000008545328>, #<Chewy::Strategy::Bypass:0x00000008544f68>, #<Chewy::Strategy::Urgent:0x0000000b80c658>]>
```

Is there a way to avoid this `:urgent` strategy getting added? 

Also, ideally I'd love to not have to worry about strategies at all when I just need to completely disable Chewy in certain environments. I'd love for there to be a configuration object like `enabled` that could be set to true just to stop everything Chewy related.

### PR DESCRIPTION
**Pull request by Vitalina-Vakulchyk**
Remove default setting of :urgent strategy for console if there is another current strategy.

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [x] ~~Updated tests.~~
* [x] Added an entry to the changelog if the new code introduces user-observable changes. See [changelog entry format](https://github.com/toptal/chewy/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CHANGELOG.md
Comment (dalthon) [inline CHANGELOG.md]:
```suggestion
    * `:urgent` is used by default if `Chewy.console_strategy` is not setted
```

File: CHANGELOG.md
Comment (dalthon) [inline CHANGELOG.md:11]:
```suggestion
    * `:urgent` is used by default if `Chewy.console_strategy` is not set
```

File: lib/chewy/config.rb
Comment (dalthon) [inline lib/chewy/config.rb:21]: It would be nice to add rdoc documentation here like we have in other options

File: lib/chewy/type/witchcraft.rb
Comment (dalthon) [inline lib/chewy/type/witchcraft.rb:226]: I am not really sure that ruby 2.5 is compatible with this syntax.
Since we say at [this part of our README](https://github.com/toptal/chewy#ruby) that we support Ruby 2.5, I guess we can't apply this.

**Review by dalthon — APPROVED** [review]
🚀 LGTM!

Just left a few comments that you can change if you like and one detail that might cause some issues with Ruby 2.5, but I will not block this PR.

**Comment by barthez** [comment]
Will it close https://github.com/toptal/chewy/issues/495?

File: .rubocop.yml
Comment (barthez) [inline .rubocop.yml:5]: As Dalton mentioned, we aim for 2.5. Please set it up here as well.

**Review by barthez — APPROVED** [review]

**Comment by Vitalina-Vakulchyk** [comment]
@barthez Yes, this PR will solve this issue https://github.com/toptal/chewy/issues/495
I am going to leave the accompanying comment there after merging.

File: .rubocop.yml
Comment (rabotyaga) [reply .rubocop.yml:5]: We check only 2.6, 2.7 & 3.0. Let's aim for 2.6 and fix the README!

**Comment by dalthon** [comment]
@Vitalina-Vakulchyk I guess @barthez asked about solving that issue because then you can follow that check on PR description: _Commit message starts with [Fix #issue-number] (if the related issue exists)._ and rename your PR description accordingly.
It helps us to keep track of PRs and Issues and I believe it automatically closes the issue once this PR is merged

File: lib/chewy/type/witchcraft.rb
Comment (dalthon) [reply lib/chewy/type/witchcraft.rb:226]: As Ivan said, the thing that is not right is the README, not the check... So fix the README and forget about this comment.

**Comment by Vitalina-Vakulchyk** [comment]
Thank you! I have changed the PR name.

**Comment by Vitalina-Vakulchyk** [comment]
Ruby 2.5 support was dropped in this PR according to discussion in the team Slack chat https://toptal-core.slack.com/archives/C6C64B7H8/p1617365634490100

File: CHANGELOG.md
Comment (rabotyaga) [inline CHANGELOG.md:18]:
```suggestion
  * [#780](https://github.com/toptal/chewy/pull/780): Changing default console strategy ([@Vitalina-Vakulchyk][]):
    * ability to set default console strategy with `Chewy.console_strategy`
    * `:urgent` is used by default if `Chewy.console_strategy` is not set
  * [#778](https://github.com/toptal/chewy/pull/778): **(Breaking)** Drop support for Ruby 2.5 ([@Vitalina-Vakulchyk][])
```

File: lib/chewy/config.rb
Comment (rabotyaga) [inline lib/chewy/config.rb:23]: Please move the doc above the setting. Lines 16-19 are about `use_after_commit_callbacks`

**Review by rabotyaga — CHANGES_REQUESTED** [review]
Please add a test

File: lib/chewy/railtie.rb
Comment (rabotyaga) [inline lib/chewy/railtie.rb:41]: Let's initialize it properly in https://github.com/toptal/chewy/pull/780/files#diff-542f2b8cbb20f056db635819b1e65c367d4e8fdb9b7e6ae000ecda0c107f1127R52-R64

File: lib/chewy/config.rb
Comment (rabotyaga) [inline lib/chewy/config.rb:17]:
```suggestion
                  # Rails console strategy, `:urgent` by default.
```

File: CHANGELOG.md
Comment (rabotyaga) [inline CHANGELOG.md:15]:
```suggestion
  * [#495](https://github.com/toptal/chewy/issues/495): Ability to change Rails console strategy with `Chewy.console_strategy` ([@Vitalina-Vakulchyk][]):
```

**Review by rabotyaga — CHANGES_REQUESTED** [review]
What about tests?

**Comment by dalthon** [comment]
@rabotyaga how do you think this should be tested?
Adding rails as a test dependency just to invoke `Rails::Console.new` like is tested at Rails itself just to capture its IO?
Do you think that it worth? Or have other idea about how to do it?

**Comment by rabotyaga** [comment]
> @rabotyaga how do you think this should be tested?
> Adding rails as a test dependency just to invoke `Rails::Console.new` like is tested at Rails itself just to capture its IO?
> Do you think that it worth? Or have other idea about how to do it?

We could add just a couple of lines to https://github.com/toptal/chewy/blob/master/spec/chewy/config_spec.rb

**Comment by Vitalina-Vakulchyk** [comment]
I added test for default console_strategy.

**Review by rabotyaga — COMMENTED** [review]
Can we test setter also?

**Comment by Vitalina-Vakulchyk** [comment]
@rabotyaga I have added setter spec.

File: spec/chewy/config_spec.rb
Comment (rabotyaga) [inline spec/chewy/config_spec.rb:101]: Should we use the `after` hook to prevent the leak of the state?

Comment (Vitalina-Vakulchyk) [reply spec/chewy/config_spec.rb]: @rabotyaga Thank you for noticing this. 
I added the `after` hook.

**Review by rabotyaga — APPROVED** [review]

**Pull request merged**
